core/src/main/java/co/aikar/commands/BaseCommand.java

```diff
@@ -531,37 +831,86 @@ public boolean canExecute(CommandIssuer issuer, RegisteredCommand<?> cmd) {
                 cmds.add(split[cmdIndex]);
             }
         }
+
         return new ArrayList<>(cmds);
     }
 
+    /**
+     * Complete a command properly per issuer and input.
+     *
+     * @param issuer
+     *         The user who executed this.
+     * @param cmd
+     *         The command to be completed.
+     * @param args
+     *         All arguments given by the user.
+     * @param commandLabel
+     *         The command name the user used.
+     * @param isAsync
+     *         Whether the command was executed async.
+     *
+     * @return All results to complete the command.
+     */
     private List<String> completeCommand(CommandIssuer issuer, RegisteredCommand cmd, String[] args, String commandLabel, boolean isAsync) {
         if (!cmd.hasPermission(issuer) || args.length > cmd.consumeInputResolvers || args.length == 0 || cmd.complete == null) {
             return ImmutableList.of();
         }
 
         List<String> cmds = manager.getCommandCompletions().of(cmd, issuer, args, isAsync);
-        return filterTabComplete(args[args.length-1], cmds);
-    }
-
-    private static List<String> filterTabComplete(String arg, List<String> cmds) {
-        return cmds.stream()
-                   .distinct()
-                   .filter(cmd -> cmd != null && (arg.isEmpty() || ApacheCommonsLangUtil.startsWithIgnoreCase(cmd, arg)))
-                   .collect(Collectors.toList());
+        return filterTabComplete(args[args.length - 1], cmds);
     }
 
+    /**
```

Comment:
again lets drop the subcommand wording.

lets rename method "getCommandBySubcommand"